File: sol/1752/1752.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        seenDrop = False
        for i in range(n - 1):
            if nums[i] > nums[i + 1]:
                if seenDrop:
                    return False
                seenDrop = True
        if seenDrop and nums[n - 1] > nums[0]:
            return False
        return True

File: sol/1752/1752.cpp
struct Solution {
    bool check(vector<int>& nums) {
        int n = nums.size();
        bool seenDrop = false;
        for (int i = 0; i < n - 1; ++i){
            if (nums[i] > nums[i + 1]){
                if (seenDrop){
                    return false;
                }
                seenDrop = true;
            }
        }
        if (seenDrop && nums.back() > nums[0]){
            return false;
        }
        return true;
    }
};
